code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -149,6 +149,12 @@
 		playsound(src, 'sound/machines/defib_zap.ogg', 50)
 		visible_message(span_warning("[src] shakes and speeds up!"))
 
+/mob/living/simple_animal/bot/secbot/handle_atom_del(atom/deleting_atom)
+	if(deleting_atom == weapon)
+		weapon = null
```

Comment:
i'd be extremely concerned if this happened but i guess this is technically right xD